custom-plugins/cms131v14_diabetes_eye_exam/protocols/cms131v14_protocol.py

```diff
@@ -0,0 +1,690 @@
+            log.info(f"CMS131v14: Patient {patient.id} has bilateral eye absence")
+            return False
+
+        return True
+
+    def _in_numerator(self, patient: Patient) -> bool:
+        if self._has_retinopathy_diagnosis_in_period(patient):
+            return self._has_retinal_exam_by_eye_care_professional_in_period(patient)
+
+        if self._has_retinal_exam_in_period_or_year_prior(patient):
+            return True
+
+        if self._has_autonomous_eye_exam_in_period(patient):
+            return True
+
+        if self._has_retinal_finding_with_severity_in_period(patient):
+            return True
+
+        if self._has_retinal_finding_no_severity_in_prior_year(patient):
+            return True
+
+        return False
+
+    def _has_diabetes_diagnosis(self, patient: Patient) -> bool:
+        try:
+            diabetes_conditions = Condition.objects.for_patient(patient.id).active()
+
+            log.info(
+                f"CMS131v14: Found {diabetes_conditions.count()} active conditions for patient {patient.id}"
+            )
+
+            for condition in diabetes_conditions:
+                for coding in condition.codings.all():
+                    if self._is_diabetes_code(coding):
+                        log.info(f"CMS131v14: Found diabetes diagnosis with code {coding.code}")
+                        return True
+
+            log.info(f"CMS131v14: No active diabetes diagnoses found for patient {patient.id}")
+            return False
+
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking diabetes diagnosis: {str(e)}")
+            return False
+
+    def _check_age_18_to_75(self, patient: Patient) -> bool:
+        try:
+            if not getattr(patient, "birth_date", None):
+                log.info("CMS131v14: Missing birth_date; patient fails age check")
+                return False
+
+            birth_date = arrow.get(patient.birth_date)
+            age_years = (self.now - birth_date).days // 365
+
+            in_range = self.AGE_RANGE_START <= age_years <= self.AGE_RANGE_END
+            log.info(
+                f"CMS131v14: Patient {patient.id} age {age_years} in 18-75 range={in_range}"
+            )
+            return in_range
+        except Exception as e:
+            log.error(f"CMS131v14: Error computing age: {str(e)}")
+            return False
+
+    def _has_diabetes_diagnosis_overlapping_period(self, patient: Patient) -> bool:
+        try:
+            diabetes_conditions = Condition.objects.for_patient(patient.id).active()
+
+            for condition in diabetes_conditions:
+                for coding in condition.codings.all():
+                    if self._is_diabetes_code(coding):
+                        return True
+
+            return False
+
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking diabetes diagnosis overlapping period: {str(e)}")
+            return False
```

Comment:
This method pulls all of the patients conditions and iterates through all the codings. Instead it should use `Condition.find(Diabetes)`